src/mapping/RadialGeoMultiscaleMapping.cpp

```diff
@@ -0,0 +1,207 @@
+#include "RadialGeoMultiscaleMapping.hpp"
+#include <Eigen/src/Core/Matrix.h>
+#include <algorithm>
+#include "mesh/Mesh.hpp"
+
+namespace precice::mapping {
+
+RadialGeoMultiscaleMapping::RadialGeoMultiscaleMapping(
+    Constraint     constraint,
+    int            dimensions,
+    MultiscaleType type,
+    MultiscaleAxis axis)
+    : Mapping(constraint, dimensions, false, Mapping::InitialGuessRequirement::None),
+      _type(type), _axis(axis)
+{
+  setInputRequirement(Mapping::MeshRequirement::VERTEX);
+  setOutputRequirement(Mapping::MeshRequirement::VERTEX);
+}
+
+void RadialGeoMultiscaleMapping::computeMapping()
+{
+  PRECICE_TRACE(output()->vertices().size());
+
+  PRECICE_ASSERT(input().get() != nullptr);
+  PRECICE_ASSERT(output().get() != nullptr);
+
+  size_t const inSize  = input()->vertices().size();
+  size_t const outSize = output()->vertices().size();
+
+  int effectiveCoordinate = _axis;
+  PRECICE_ASSERT(effectiveCoordinate == 0 ||
+                     effectiveCoordinate == 1 ||
+                     effectiveCoordinate == 2,
+                 "Unknown multiscale axis type.")
+
+  // determine principle axis midpoints as borders to assign 3D vertices to their respective 1D vertex
+  Eigen::VectorXd axisMidpoints(inSize);
+  _axisMidpoints        = axisMidpoints;
+  auto ordered_vertices = input()->vertices();
```

Comment:
Different comment: Does this give us a copy of the vertices or are we modifying the original? Not sure how bad this really is, but I have a bad gut feeling if we sort the original vertices.
@fsimonis you too?

Reply:
> Is it correct that always input mesh is the 1D mesh? This should only be the case in a spread mapping, no?

You're right! I corrected it, it now reads `auto ordered_vertices = (_type == SPREAD) ? input()->vertices() : outputput()->vertices();`

> Different comment: Does this give us a copy of the vertices or are we modifying the original? Not sure how bad this really is, but I have a bad gut feeling if we sort the original vertices. @fsimonis you too?

Good question, I operated under the assumption that it'd give me a copy, because I too would be uncomfortable with ordering the original vertices. Since the `vertices()` method of the `mesh` class returns a `Mesh::VertexContainer` reference, I'm afraid I might be ordering the original vertices. However, when I try to copy-construct another `ordered_vertices` like this 
```
auto vertices = (_type == SPREAD) ? input()->vertices() : output()->vertices();
precice::mesh::Mesh::VertexContainer ordered_vertices(vertices);
```
I get a runtime error stating 
```
/usr/include/c++/9/bits/stl_deque.h:1402: std::deque<_Tp, _Alloc>::reference std::deque<_Tp, _Alloc>::operator[](std::deque<_Tp, _Alloc>::size_type) [with _Tp = precice::mesh::Vertex; _Alloc = std::allocator<precice::mesh::Vertex>; std::deque<_Tp, _Alloc>::reference = precice::mesh::Vertex&; std::deque<_Tp, _Alloc>::size_type = long unsigned int]: Assertion '__builtin_expect(__n < this->size(), true)' failed.
```

Reply:
> Does this give us a copy of the vertices or are we modifying the original?

This copies the deque of vertices.

It is pretty harsh to copy the complete vertices of a mesh though.
My suggestion:
* Use contiguous storage `std::vector` instead of the `std::deque`
* Sort indices or pointers instead of vertices

Also use `Vertex::getRawCoords()` as `Vertex::getCoords()` is an allocating call.

Reply:
@fsimonis Updated the sorting. Please check again